src/langchain_google_alloydb_pg/model_manager.py

```diff
@@ -0,0 +1,202 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from dataclasses import dataclass
+from typing import Any, List, Sequence
+
+from sqlalchemy import text
+from sqlalchemy.engine.row import RowMapping
+from sqlalchemy.ext.asyncio import AsyncEngine
+
+
+@dataclass
+class AlloyDBModel:
+    model_id: str
+    model_request_url: str
+    model_provider: str
+    model_type: str
+    model_qualified_name: str
+    model_auth_type: str
+    model_auth_id: str
+    input_transform_fn: str
+    output_transform_fn: str
+
+
+class AlloyDBModelManager:
+    """Manage models to be used with google_ml_integration Extension.
+    Refer to [Model Endpoint Management](https://cloud.google.com/alloydb/docs/ai/model-endpoint-overview).
+    """
+
+    def __init__(
+        self,
+        engine: AsyncEngine,
+    ):
+        """AlloyDBModelManager constructor.
+        Args:
+            engine (Asycn AlloyDBEngine): Connection pool engine for managing connections to Postgres database.
+        """
+
+        self._engine = engine
+
+        self._engine._run_as_sync(self.__validate())
+
+    async def alist_model(
+        self, model_id: str = "textembedding-gecko@003"
+    ) -> AlloyDBModel:
+        """Lists the model details for a specific model_id.
+
+        Raises:
+            :class:`DBAPIError <sqlalchemy.exc.DBAPIError>`: if model has not been created.
+        """
+        result = await self._engine._run_as_async(self.__alist_model(model_id=model_id))
+        return result
+
+    async def amodel_info_view(self) -> List[AlloyDBModel]:
+        """Lists all the models and its details."""
+        results = await self._engine._run_as_async(self.__amodel_info_view())
+        return results
+
+    async def acreate_model(
+        self, model_id: str, model_provider: str, **kwargs: dict[str, Any]
```

Comment:
Here and in other places, kwargs can be of type `dict[str, str]`. The reference doc doesn't yet specific any non-string field.